### PR TITLE
Update GraphQL library reference

### PR DESCRIPTION
**Pull request by rodolfotorres**
## Description
This PR updates the GraphQL library version to v2.1 so it can be out of alpha.

## Notes and considerations for @tlil
* I had to fix 2 tests that I would like a second opinion on
* If I used the engine definition of `UriGraphType` then the tests on `UriGraphTypeTests` would actually break so I decided to keep it

### CODE REVIEW

File: test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public async Task Can_Execute_Query_On_Schema_From_Abstract_Types()
                 .Execute();
 
             result.ShouldHaveNoErrors();
-            result.Data.ShouldHaveFieldWithValue("commonField", new DateTime(1970, 1, 1));
+            result.Data.ShouldHaveFieldWithValue("commonField", "1970-01-01");
```
Comment (huysentruitw) [inline test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs:34]: Having to change a unit-test like that should ring a bell. The fact that the `CommonField` object is no longer a `DateTime` but a `string` in the newer version sounds like a bug that needs to be fixed.

File: src/GraphQL.Conventions/Extensions/Profiling/ProfilingResultEnricher.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public static void EnrichWithProfilingInformation(this Response response)
 
                 records.Add(new PerformanceRecord
                 {
-                    Path = record.Metadata.ContainsKey("path") ? string.Join(".", record.Metadata["path"] as List<string>) : null,
+                    Path = record.Metadata.ContainsKey("path") ? string.Join(".", record.Metadata["path"] as string[]) : null,
```
Comment (huysentruitw) [inline src/GraphQL.Conventions/Extensions/Profiling/ProfilingResultEnricher.cs]: Is this change really needed?

File: test/Tests/Adapters/TypeMappingTests.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public void Can_Derive_Nullable_Primitive_Types()
             Type<TimeSpan?>().ShouldBeOfType<TimeSpanGraphType>();
             Type<Id?>().ShouldBeOfType<Extended.IdGraphType>();
             Type<Url>().ShouldBeOfType<UrlGraphType>();
-            Type<Uri>().ShouldBeOfType<UriGraphType>();
+            Type<Uri>().ShouldBeOfType<ConventionsTypes.UriGraphType>();
```
Comment (huysentruitw) [inline test/Tests/Adapters/TypeMappingTests.cs]: Can you put a `using UriGraphType = GraphQL.Conventions.Adapters.Types.UriGraphType;` on top instead of changing this in multiple places? Currently, you're picking the wrong type.

File: test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs (hunk 1 above)
Comment (huysentruitw) [reply test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs:34]: I tried to step through the code for the different versions, but it's unclear where things go wrong :)

**Comment by huysentruitw** [comment]
Let me link this issue here https://github.com/graphql-dotnet/graphql-dotnet/issues/846

File: test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs (hunk 1 above)
Comment (huysentruitw) [reply test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs:34]: The failing test here is caused by the introduction of new DateTimeGraphType and DateTimeOffsetGraphType. I've created an other pull-request that includes these changes and updates to version 2.1.0, so this PR can be closed.

See #106

**Comment by huysentruitw** [comment]
The failing test here is caused by the introduction of new DateTimeGraphType and DateTimeOffsetGraphType. I've created an other pull-request that includes these changes and updates to version 2.1.0, so this PR can be closed.

See #106

File: test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs (hunk 1 above)
Comment (Igorbek) [inline test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs:34]: FYI, this tracks down to the following change in `graphql-dotnet`: https://github.com/graphql-dotnet/graphql-dotnet/pull/665/files#diff-6ecb3c17fef816d6cd02a67ad408ea9eR19

**Review by Igorbek — APPROVED** [review]

File: test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs (hunk 1 above)
Comment (Igorbek) [reply test/Tests/Adapters/Engine/AbstractTypeConstructionTests.cs:34]: the change in `graphql-dotnet`: https://github.com/graphql-dotnet/graphql-dotnet/pull/665/files#diff-6ecb3c17fef816d6cd02a67ad408ea9eR19

**Comment by huysentruitw** [comment]
Can be closed. Replaced by #106

**Comment by tlil** [comment]
Thanks. Closing as duplicate of #106 which is now merged.
